File: src/main/java/hu/indicium/dev/services/events/api/commands/CreateEventCommand.java
package hu.indicium.dev.services.events.api.commands;

import lombok.Data;

import javax.validation.constraints.*;
import java.time.LocalDateTime;

@Data
public class CreateEventCommand {
    @NotBlank
    @Size(min = 1, max = 100)
    private String title;

    @NotBlank
    @Size(min = 1, max = 100)
    private String description;

    @NotBlank
    @Size(min = 1, max = 100)
    private String slug;

    @NotNull
    private LocalDateTime startDateTime;

    @NotNull
    private LocalDateTime endDateTime;
}

File: src/main/kotlin/hu/indicium/dev/services/events/api/commands/CreateEventCommand.kt
package hu.indicium.dev.services.events.api.commands

import java.time.LocalDateTime
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

data class CreateEventCommand(
    @get:NotBlank
    @get:Size(min = 1, max = 100)
    val title: String? = null,

    @get:NotBlank
    @get:Size(min = 1, max = 100)
    val description: String? = null,

    @get:NotBlank
    @get:Size(min = 1, max = 100)
    val slug: String? = null,

    @get:NotNull
    val startDateTime: LocalDateTime? = null,

    @get:NotNull
    val endDateTime: LocalDateTime? = null
)